### DISCUSSION TITLE
ftdi

### DISCUSSION THREAD

**Discussion by austinandya** [nathanmarlor/foxess_modbus#467] [General]
Does this driver support a USB adaptor based on the FTDi chipset

**Comment by canton7** [comment]
It doesn't care. As long as the OS presents it as a serial device, it can use it as such.

**Reply by austinandya** [reply]
Thanks I have a Foxess A10 series inverter. I can see the ttyUSB0 with this id /dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A50285BI-if00-port0 but the FOXESS Inverter does not identify the target device at address 247. Not something you may know how to solve but worth a try. Regards Andrew

**Reply by canton7** [reply]
By "A10" do you mean "AIO" (all-in-one)?

What error message exactly do you get?

**Comment by austinandya** [comment]
Hi Nathan

 

Yes correct AIO with this response.

**Reply by canton7** [reply]
You need to post images using the github web interface: attaching them to emails doesn't work.

**Reply by austinandya** [reply]
Sorry sending from email.
![Capture 1](https://github.com/nathanmarlor/foxess_modbus/assets/33328611/efb52136-a820-4bef-9771-d62a7dde6e5f)

**Reply by austinandya** [reply]
Also this details on USB0
![Capture 2](https://github.com/nathanmarlor/foxess_modbus/assets/33328611/31c39b7d-74e2-4c81-9f28-c4c70ed27348)

**Reply by canton7** [reply]
Aah, you've hit #457. Try the latest pre-release, see if that helps.

**Comment by austinandya** [comment]
Hi Nathan. This is the USB0 setting.

 



 

Andrew

**Comment by austinandya** [comment]
Yes I got that once I went back to the feed. I am looking at the RS485 signals across 100Ohm resistor and the levels do not look great need to do further investigation on that.

 

Tanks for the help up to now.

 

Andrew

**Reply by austinandya** [reply]
They are at 1V PP so good enough. Looks like another issue. No response form unit.
Thanks

**Reply by canton7** [reply]
I'm confused. The error you were getting would stop the adapter working at all. Has the error message changed?

**Reply by austinandya** [reply]
Me too Nathan, No change to response. I will try doing a serarate test first using dedicated micro to see if I get a response to a RS485 command using a stand alone solution without all the other systems. File descriptor out of range does not make sense to me.

Chat again if you dont mind. Thanks.

**Reply by canton7** [reply]
(I'm not Nathan btw).

File descriptor out of range means your system has allocated more than 1024 file descriptors, and there are some syscalls which don't support fds > 1024. Restarting will often fix the problem temporarily. As I said, the latest pre-release attempts to work around the problem.

**Reply by austinandya** [reply]
Thanks Antony. I had to rush off. will try this again soon.

**Reply by austinandya** [reply]
I removed foxess modbus and reinstalled with pre-release the latest and tried again to talk to AIO inverter. Different response. Also tried to restart Home Assistant.

Error connecting to your adapter or inverter. Ensure the adapter is properly configured and is correctly wired to your inverter (see the setup link above), then try again. Details: cannot access local variable 'fd' where it is not associated with a value

Any comments on this one.

**Reply by canton7** [reply]
Looks like this is a bug in pyserial.

I've had a go at working around it Can you give v1.8.0b5 a try please?

**Comment by austinandya** [comment]
Hi Antony

 

I will give it a go. Just give me some time as I have a big week ahead.

 

Thanks Kindly
